File: patterns-use-cases/integrations/java-spring-maven/src/main/java/com/example/demo/Greeter.java
package com.example.demo;

import dev.restate.sdk.Context;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.Service;
import org.springframework.stereotype.Component;

@Component
@Service
public class Greeter {

    @Handler
    public String greet(Context ctx, String greeting) {
        return greeting;
    }
}

File: templates/kotlin-gradle/src/main/kotlin/my/example/Greeter.kt
package my.example

import dev.restate.sdk.annotation.Handler
import dev.restate.sdk.annotation.Service
import dev.restate.sdk.http.vertx.RestateHttpEndpointBuilder
import dev.restate.sdk.kotlin.Context

/**
 * Template of a Restate service and handler
 * Have a look at the Kotlin QuickStart to learn how to run this: https://docs.restate.dev/get_started/quickstart?sdk=kotlin
 */
@Service
class Greeter {

  @Handler
  suspend fun greet(ctx: Context, greeting: String): String {

    return greeting
  }
}

fun main() {
  RestateHttpEndpointBuilder
          .builder()
          .bind(Greeter())
          .buildAndListen()
}